tests/gis_tests/relatedapp/tests.py

```diff
@@ -304,6 +306,38 @@ def test_collect(self):
         self.assertEqual(4, len(coll))
         self.assertTrue(ref_geom.equals(coll))
 
+    @skipUnlessDBFeature("supports_collect_aggr")
+    def test_collect_filter(self):
+        city = City.objects.get(name="Aurora")
+        city.snack_set.create(point=Point(1, 2), name="Vegan Pizza")
+        city.snack_set.create(point=Point(3, 4), name="Sandwich")
+        city.snack_set.create(point=Point(4, 5), name="Vegan Smoothies")
+        city2 = City.objects.get(name="Roswell")
+        city2.snack_set.create(point=Point(1, 2), name="Pizza Party")
+        city2.snack_set.create(point=Point(2, 3), name="Nothing")
+        city3 = City.objects.get(name="Dallas")
+        city3.snack_set.create(point=Point(2, 3), name="Nothing interesting")
+        city3.snack_set.create(point=Point(7, 8), name="Vegan Chili")
+
+        q_filter = Q(snack__name__startswith="Vegan")
+        annotated_queryset = City.objects.annotate(
+            snack_points=Collect("snack__point", filter=q_filter),
+            snack_center=Centroid(Collect("snack__point", filter=q_filter)),
+        )
+        aurora = annotated_queryset.get(name="Aurora")
+        self.assertEqual(aurora.snack_points.wkt, "MULTIPOINT (1 2, 4 5)")
+        self.assertEqual(aurora.snack_center.wkt, "POINT (2.5 3.5)")
+        kecksburg = annotated_queryset.get(name="Kecksburg")
+        self.assertEqual(kecksburg.snack_points, None)
+        self.assertEqual(kecksburg.snack_center, None)
+        roswell = annotated_queryset.get(name="Roswell")
+        self.assertEqual(roswell.snack_points, None)
+        self.assertEqual(roswell.snack_center, None)
+        dallas = annotated_queryset.get(name="Dallas")
+        # Postgis keep a MULTIPOINT but Spatialite collapse it as a point.
+        self.assertIn(dallas.snack_points.wkt, ("MULTIPOINT (7 8)", "POINT (7 8)"))
+        self.assertEqual(dallas.snack_center.wkt, "POINT (7 8)")
```

Comment:
Do we need a new model and objects? We should try to reuse existing cities e.g.
```python
    @skipUnlessDBFeature("supports_collect_aggr")
    def test_collect_filter(self):
        ref_geom = GEOSGeometry(
            "MULTIPOINT(-97.516111 33.058333,-96.801611 32.782057,"
            "-95.363151 29.763374,-96.801611 32.782057)"
        )   
        qs = City.objects.aggregate(
            points=Collect("location__point", filter=Q(state="TX"))
        )   
        collected_points = qs["points"]
        self.assertEqual(len(collected_points), 4)
        self.assertIs(ref_geom.equals(collected_points), True)
```
Can we add small test methods for other GIS aggregates?

Reply:
There is only one Location per City so I didn't think that it made much sense for theses tests.

I  can try using Parcel but there it require lot of arguments and they are no fixtures for them.
Will add tests for the other aggregate